status/discover.go

```diff
@@ -91,66 +85,89 @@ func getPathURL(endpoint string) (path string) {
 	return u.Path
 }
 
-func checkTracingService() (url string) {
-	tracingConfig := config.Get().ExternalServices.Tracing
-	service := tracingConfig.Service
-
-	if service != "" {
-		// Try to discover the URL
-		url = discoverServiceURL(tracingConfig.Namespace, service)
-	} else {
-		// Check usual route names for tracing
-		for _, name := range tracingLookupRoutes {
-			service = name
-			url = discoverServiceURL(tracingConfig.Namespace, name)
-			if url != "" {
-				break
-			}
-		}
+func discoverTracingService() (service string) {
+	client, err := getClient()
+	//  Return if there is a problem with the client
+	if err != nil {
+		log.Debugf("[TRACING] Service discovery failed: %v", err)
+		return
 	}
-
-	// The user has set the route or we found one in tracing or jaeger-query
-	if url != "" {
-		// Calculate if Path
-		path, err := checkIfQueryBasePath(tracingConfig.Namespace, service)
-		if err != nil {
-			log.Debugf("Error checking the query base path")
-		}
-		// The user didn't set the URL, so we need to set
-		if tracingConfig.URL == "" {
-			tracingConfig.URL = url + path // Overwrite URL if the user didn't set
+	// Check for each service in list
+	for _, name := range tracingLookupRoutes {
+		service = name
+		// Try to discover the service
+		serv, err := client.GetService(config.Get().IstioNamespace, service)
+		// If there is no error and the service is not nil that means that we found tracing
+		if serv != nil && err == nil {
+			log.Debugf("[TRACING] Service in: %s", service)
+			break
+		} else {
+			// No service found set to empty for the last iteration
+			service = ""
 		}
-
-		// We store the path
-		tracingConfig.Path = path
-
-		// Set the service
-		tracingConfig.Service = service
-		appstate.JaegerEnabled = true
 	}
+	return
+}
 
-	// Save config
-	appstate.JaegerConfig = tracingConfig
+func discoverTracingPath() (path string) {
+	tracingConfig := config.Get().ExternalServices.Tracing
+	// We had the service so we can check the Path
+	path, err := checkIfQueryBasePath(tracingConfig.Namespace, tracingConfig.Service)
+	if err != nil {
+		log.Debugf("[TRACING] Error checking the query base path")
+	}
+	return
+}
 
-	return tracingConfig.URL
+func discoverURLTracingService() (url string) {
+	// Try to discover the URL . Openshift Client
+	url, err := discoverServiceURL(appstate.JaegerConfig.Namespace, appstate.JaegerConfig.Service)
+	if err != nil {
+		log.Debugf("[TRACING] URL discovery failed: %v", err)
+		return
+	}
+	// Trim the string to format correctly the url with the path
+	url = strings.TrimSuffix(url, "/") + "/" + appstate.JaegerConfig.Path
```

Comment:
If `err` was empty but `url` is empty too, we still run through this code, which seems to be wrong. This case happens when we're not on openshift.

Reply:
Right, It's not necessary print the log neither because we did in the discovery method.